content/modules/python/pages/writing_functions.adoc

```diff
@@ -0,0 +1,462 @@
+
+[source, Python]
+----
+def sum_then_multiply_by_x(x, *args) -> int:
+    return sum(args)*x
+
+sum_then_multiply_by_x(1,2,3,4,5)
+----
+
+[source, Python]
+----
+14
+----
+
+To create one or more _positional_ only arguments, simply add a `/` as a standalone argument _after_ all of the arguments which you would like to be _only_ positional. For example:
+
+[source, Python]
+----
+def sum_then_multiply_by_x(one, two, /, three, x) -> int:
+    return sum([one, two, three])*x
+
+sum_then_multiply_by_x(1,2,3,4) # all positional, will work
+sum_then_multiply_by_x(1,2,three=3,x=4) # two keyword, two positional, will work
+sum_then_multiply_by_x(1,two=2,three=3,x=4) # a positional only argument was passed as a keyword argument, error
+----
+
+== docstrings
+
+docstrings are the strings inside the function immediately following the function declaration. docstrings provide documentation for the function. You can put any information you'd like in a docstring, however, it is best to carefully describe what the function does, and stay consistent in style from docstring to docstring.
+
+You can access a function's docstring in various ways:
+
+[source, Python]
+----
+print(word_count.__doc__)
+----
+
+[source, Python]
+----
+word_count is a function that accepts a sentence as an argument,
+and returns the number of words in the sentence.
+ 
+Args:
+sentence (str): The sentence for which we are counting the words.
+ 
+Returns:
+int: The number of words in the sentence
+ 
+----
+
+[source, Python]
+----
+help(word_count)
+----
+
+[source, Python]
+----
+Help on function word_count in module __main__:
+ 
+word_count(sentence: str) -> int
+word_count is a function that accepts a sentence as an argument,
+and returns the number of words in the sentence.
+     
+Args:
+sentence (str): The sentence for which we are counting the words.
+     
+Returns:
+int: The number of words in the sentence
+----
+
+In addition, if you are coding in a tool like VSCode, for example, you may gain the ability to hover over a function and see its docstring and other information.
+
+It is good practice to write docstrings for every function you write.
+
+== annotations
+
+Another "thing" you may have noticed from our `word_count` function if you've ever used Python in the past. In the _signature_ of our function `def` `word_count(sentence: str) -> int`, we have some extra, _not_ required information in the form of _function annotations_. Specifically, you could write the `word_count` function like this, but it would function just the same:
```

Comment:
In the `word_count` example, you may notice that the format is a little bit different from the rest of the examples used on this page. The `word_count` function consists extra (not necessary) information. This is a form of type annotations. 

You can shorten the `word_count` function, and it still will do the same: